Validation administration Client

remycool opened the pull request:
Tests à réaliser

Commit:
Ajout du view model Clientet des  règles de validations

remycool committed:
Validation des données formulaire client

remycool committed:
Ajout vue client edit

remycool committed:
Renommage VueModele

remycool committed:
merge master dans feature_client

remycool committed:
Renommage des vues client, application du MVVM pour VueClientList

remycool committed:
Ajout de VueModeleClientEdit

remycool committed:
TypeAhead sur le champs code postal

Commit:
Validation des donnees et possiblité d'enregistrer un client depuis l'IHM

Commit:
Navigation vers page précédente opérationnelle

Commit:
Vue Client Edit
 --> Redirection vers page précédente et logout
Vue ClientList
--> Redirection vers logout

Commit:
update client  non fonctionnel

Commit:
Affichage du client selectionne dans la vue ClientEdit Ok

Commit:
Update depuis vue ClientEdit ok
Recherche étendue à toutes les colonnes de la grille dans la vue ClientList
Ajout d'une check box dans ClientEdit pour déterminer le statut client

Commit:
Factorisation dans clientDAL
Remplacement chexkbox dans vue ClientEdit
Grille non modifiable dans vue ClientList

Commit:
Correction bug "switch toggle" dans la vue ClientEdit à la modification d'un client
 et documentation du code

remycool commented:
Tests unitaires fait

lowxorx commented:
Besoin de remettre la bdd mysql en ligne pour les tests

Commit:
Remplacement du nom de la Bdd pour les requêtes
Modification de la connexion string MySql

lowxorx commented:
L'exception MySql.Data.MySqlClient.MySqlException s'est produite
  HResult=0x80004005
  Message=Table 'housemadera.commune' doesn't exist
  Source=MySql.Data
  Arborescence des appels de procédure :
   à MySql.Data.MySqlClient.MySqlStream.ReadPacket()
   à MySql.Data.MySqlClient.NativeDriver.GetResult(Int32& affectedRow, Int64& insertedId)
   à MySql.Data.MySqlClient.Driver.GetResult(Int32 statementId, Int32& affectedRows, Int64& insertedId)
   à MySql.Data.MySqlClient.Driver.NextResult(Int32 statementId, Boolean force)
   à MySql.Data.MySqlClient.MySqlDataReader.NextResult()
   à MySql.Data.MySqlClient.MySqlCommand.ExecuteReader(CommandBehavior behavior)
   à MySql.Data.MySqlClient.MySqlCommand.ExecuteDbDataReader(CommandBehavior behavior)
   à System.Data.Common.DbCommand.ExecuteReader()
   à HouseMadera.DAL.MySqlConnect.GetDataReader(DbCommand command) dans C:\Users\Lowxorx\Source\Repos\HouseMadera\HouseMadera\HouseMadera\DAL\MySqlConnect.cs :ligne 60
   à HouseMadera.DAL.DAL.GetDataReader(DbCommand command) dans C:\Users\Lowxorx\Source\Repos\HouseMadera\HouseMadera\HouseMadera\DAL\DAL.cs :ligne 90
   à HouseMadera.DAL.DAL.Get(String requete, IDictionary`2 parameters) dans C:\Users\Lowxorx\Source\Repos\HouseMadera\HouseMadera\HouseMadera\DAL\DAL.cs :ligne 98
   à HouseMadera.DAL.CommuneDAL.GetFilteredCommunes(Int32 value) dans C:\Users\Lowxorx\Source\Repos\HouseMadera\HouseMadera\HouseMadera\DAL\Commune\CommuneDAL.cs :ligne 28
   à HouseMadera.VueModele.VueModeleClientEdit.RechercherCommunes(String codePostal) dans C:\Users\Lowxorx\Source\Repos\HouseMadera\HouseMadera\HouseMadera\VueModele\VueModeleClientEdit.cs :ligne 458
   à HouseMadera.VueModele.VueModeleClientEdit.set_CodePostal(String value) dans C:\Users\Lowxorx\Source\Repos\HouseMadera\HouseMadera\HouseMadera\VueModele\VueModeleClientEdit.cs :ligne 98
   à HouseMadera.VueModele.VueModeleClientEdit.InitVueModele(Client clientSelectionne) dans C:\Users\Lowxorx\Source\Repos\HouseMadera\HouseMadera\HouseMadera\VueModele\VueModeleClientEdit.cs :ligne 368
   à HouseMadera.VueModele.VueModeleClientList.ModifierClient() dans C:\Users\Lowxorx\Source\Repos\HouseMadera\HouseMadera\HouseMadera\VueModele\VueModeleClientList.cs :ligne 169

Polotoz commented:
Tests :
Exception sur la table Commune qui n'existe pas sur MySQL
Lors d'un enregistrement d'un client (modification ou ajout) rediriger vers l'écran avec la grid
Rendre obligatoire un seul numéro de téléphone (si l'on renseigne le premier numéro on ne peut pas enregistrer, mais si on clique sur l'autre champ numéro on peut enregistrer).

Commit:
Correctif suite pull request #5:
- MySQL Exception : la methode d'accés aux communes a été entouré d'un try-catch
- Redirection vers grille client après validation (vue d'edition du client)
- Redirection vers accueil depuis vue administration client

Commit:
resolutions de conflit

lowxorx commented:
Suite dernier commit, tests OK
Merge dans master
